exporter/sapmexporter/spanshims.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sapmexporter
+
+import (
+	"github.com/signalfx/signalfx-agent/pkg/apm/tracetracker"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+var (
+	_ tracetracker.SpanList = (*spanListWrap)(nil)
+	_ tracetracker.Span     = (*spanWrap)(nil)
+)
+
+type spanWrap struct {
+	pdata.ResourceSpans
+}
+
+func (s spanWrap) Environment() (string, bool) {
+	res := s.Resource()
+	if res.IsNil() {
+		return "", false
+	}
+	attr := res.Attributes()
+
+	// Try to find deployment.environment before falling back to environment (SignalFx value).
+	env, ok := attr.Get(conventions.AttributeDeploymentEnvironment)
+	if ok && env.StringVal() != "" {
+		return env.StringVal(), true
+	}
+
+	env, ok = attr.Get("environment")
+	if ok && env.StringVal() != "" {
+		return env.StringVal(), true
+	}
+
+	return "", false
+}
+
+func (s spanWrap) ServiceName() (string, bool) {
+	res := s.Resource()
+	if res.IsNil() {
+		return "", false
+	}
+	attr := res.Attributes()
+
+	serviceName, ok := attr.Get(conventions.AttributeServiceName)
+	if ok && serviceName.StringVal() != "" {
+		return serviceName.StringVal(), true
+	}
+
+	return "", false
+}
+
+func (s spanWrap) Tag(tag string) (string, bool) {
+	res := s.Resource()
+	if res.IsNil() {
+		return "", false
+	}
+	attr := res.Attributes()
```

Comment:
Do otel trace spans not have attributes on the spans themselves or is it all on Resource?

Reply:
Yeah spans can have attributes (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/) but we're only interested in environment & service which are only on the resource.